applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -77,14 +79,96 @@ namespace Kratos {
+            {
+                rVector.resize(3);
+                std::size_t index = 0;
+                Node<3>& r_node = mpElement->GetGeometry()[NodeId];
+                rVector[index++] = MakeIndirectScalar(r_node, DISPLACEMENT_X, Step);
+                rVector[index++] = MakeIndirectScalar(r_node, DISPLACEMENT_Y, Step);
+                rVector[index] = MakeIndirectScalar(r_node, DISPLACEMENT_Z, Step);
+            }
+
+            void GetFirstDerivativesVector(std::size_t NodeId,
+                                           std::vector<IndirectScalar<double>>& rVector,
+                                           std::size_t Step,
+                                           ProcessInfo& rCurrentProcessInfo) override
+            {
+                rVector.resize(4);
+                std::size_t index = 0;
+                Node<3>& r_node = mpElement->GetGeometry()[NodeId];
+                rVector[index++] = MakeIndirectScalar(r_node, VELOCITY_X, Step);
+                rVector[index++] = MakeIndirectScalar(r_node, VELOCITY_Y, Step);
+                rVector[index++] = MakeIndirectScalar(r_node, VELOCITY_Z, Step);
+                rVector[index] = MakeIndirectScalar(r_node, PRESSURE, Step);
+            }
+
+            void GetSecondDerivativesVector(std::size_t NodeId,
+                                            std::vector<IndirectScalar<double>>& rVector,
+                                            std::size_t Step,
+                                            ProcessInfo& rCurrentProcessInfo) override
+            {
+                rVector.resize(3);
+                std::size_t index = 0;
+                Node<3>& r_node = mpElement->GetGeometry()[NodeId];
+                rVector[index++] = MakeIndirectScalar(r_node, ACCELERATION_X, Step);
+                rVector[index++] = MakeIndirectScalar(r_node, ACCELERATION_Y, Step);
+                rVector[index] = MakeIndirectScalar(r_node, ACCELERATION_Z, Step);
+            }
+
+            void GetFirstDerivativesDofsVector(std::size_t NodeId,
+                                               std::vector<Dof<double>::Pointer>& rVector,
+                                               ProcessInfo& rCurrentProcessInfo) override
+            {
+                rVector.resize(4);
+                std::size_t index = 0;
+                Node<3>& r_node = mpElement->GetGeometry()[NodeId];
+                rVector[index++] = r_node.pGetDof(VELOCITY_X);
+                rVector[index++] = r_node.pGetDof(VELOCITY_Y);
+                rVector[index++] = r_node.pGetDof(VELOCITY_Z);
+                rVector[index] = r_node.pGetDof(PRESSURE);
+            }
+
+            void GetZeroDerivativesVariables(std::vector<VariableData const*>& rVariables,
+                                             ProcessInfo& rCurrentProcessInfo) const override
+            {
+                rVariables.resize(1);
+                rVariables[0] = &DISPLACEMENT;
```

Comment:
does this work?
Afaik there are currently some problems with the storage of variables

Reply:
This seems to be working :)